File: zircon/tools/fidl/lib/check.cc
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "fidl/check.h"

#include <stdio.h>
#include <stdlib.h>

namespace fidl {

void LogMessageAndAbort(const char* file, int line, const char* condition, const char* message) {
  fprintf(stderr, "Check failed: %s\n%s:%d: %s\n", condition, file, line, message);
  abort();
}

}

File: tools/update_crates/src/check.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use argh::FromArgs;
use std::path::PathBuf;

/// update out of date crates
/// check that updates did not violate update policies
#[derive(Debug, FromArgs)]
#[argh(subcommand, name = "check")]
pub struct CheckOptions {}

pub fn check_for_new_crates(_manifest_path: PathBuf) -> anyhow::Result<()> {
    println!("WARNING: check for newly added dependencies and get OSRB approval before uploading.");
    Ok(())
}
